### PR TITLE
WIP: Implement PeerManager

### PR DESCRIPTION
**Pull request by gsalgado**
This kind of works currently, but it is very hard to find les/1 nodes on the same
network_id that don't have too many peers yet.

### CODE REVIEW

**Comment by pipermerriam** [comment]
@gsalgado would it help for you to have a dedicated LES node running somewhere that you can connect to?

**Comment by gsalgado** [comment]
I have one running locally, which I use to test the Peer, but the PeerManager needs to be tested against the actual P2P network, to check that it can find useful nodes and connect to them. That works, but there seem to be few nodes out there that support LES/1 *and* don't have too many peers connected already, so most of the time it takes a while to find even just one of them. AIUI, the v5 of the discovery protocol is supposed to solve this

File: evm/p2p/peermanager.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    discovery_sleep = 3
+
+    def __init__(self, discovery):
+        self.discovery = discovery
+        self.connected_nodes = {}
+        # FIXME: Need a persistent DB to keep track of failed/successful peers or else every time
+        # we start it takes ages to find peers that support les/1 and are on the same network as
+        # us.
+        self.failed_remotes = []
+        self.useless_peers = []
```
Comment (pipermerriam) [inline evm/p2p/peermanager.py]: And a set for this as well?

**Comment by gsalgado** [comment]
The `PeerPool` class, which was merged recently, supersedes this

**Pull request closed without merging**